messaging.rst

```diff
@@ -104,38 +104,113 @@ Metadata
+Note the encoding of the encrypted object's field is not part of this specification.
+
+When the target receives a transfer with an encrypted secret, it may try to decrypt it::
+
+        { "secret": <bytes32>, "amount": <int32>, "payment_identifier": <int32> } = encrypt_ecies(target_privatekey, secret)
+
+
+The target should only accept the decrypted secret, if the decrypted objects fields comply to the following properties:
+
+- ``secret`` must hash to the LT's ``lock.secrethash``
+- ``amount`` must be less than or equal to the received LT's ``lock.amount``
+- ``payment_identifier`` must be equal to LT's ``payment_identifier``
+
+If the validation passes, the target uses "secret" as this transfer's secret and skips sending a :ref:`SecretRequest <secret-request-message>` to the initator.
+Therefore the target can immediately start the unlock phase by sending a corresponding :ref:`RevealSecret <reveal-secret-message>` to the last mediator in the transfer's route.
+
+The Metadata should be determined by the initiator. All mediators of a transfer should pass it to the next hop of the transfer as-is and
+should not modify it. Complying to this behaviour is advertised as the :ref:`capability <transport-capabilities>` `immutableMetada=1`.
 
 Fields
 ^^^^^^
 
-+----------------------+---------------------+----------------------------+
-| Field Name           | Field Type          |  Description               |
-+======================+=====================+============================+
-|  routes              | List[RouteMetadata] | A list of possible routes. |
-+----------------------+----------------+---------------------------------+
++----------------------+------------------------------+---------------------------------------------------------+
+| Field Name           | Field Type                   | Description                                             |
++======================+==============================+=========================================================+
+|  routes              | List[RouteMetadata]          | A list of possible routes and additional route metadata |
++----------------------+------------------------------+---------------------------------------------------------+
+|  secret              | Optional[bytes32]            | The LT's secret encrypted with the target's public key  |
++----------------------+------------------------------+---------------------------------------------------------+
 
 RouteMetadata
 -------------
 
 .. _route-metadata:
 
-This datastructure describes a single route, the noce receiving the message
-will be the first entry in the list, the second node is the node which the
-transfer should be forwarded to.
+This datastructure describes a single route and the metadata for all nodes included in that route.
 
-Each entry is EIP55-checksum addresses with ``0x``-prefix as usual. The last of
+Each entry in the ``route`` field list is EIP55-checksum addresses with ``0x``-prefix as usual. The last of
```

Comment:
We had removed address checksum format as a hard-requirement on the protocol, and instead coded the clients to accept any format plus as-is signature verification, which would allow any supported format to interoperate. Should we adapt the spec here?

Reply:
Yes, this makes sense